src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxEquivalenceTests.cs

```diff
@@ -448,5 +455,112 @@ public void TestExpressionBodiedMethod()
             VerifyEquivalent(tree1, tree2, topLevel: true);
             VerifyNotEquivalent(tree1, tree2, topLevel: false);
         }
+
+        [Theory, WorkItem(38694, "https://github.com/dotnet/roslyn/issues/38694")]
+        [InlineData("#nullable enable", "#nullable disable")]
+        [InlineData("#nullable enable", "#nullable restore")]
+        [InlineData("#nullable disable", "#nullable restore")]
+        [InlineData("", "#nullable disable")]
+        [InlineData("", "#nullable enable")]
+        [InlineData("", "#nullable restore")]
+        public void TestNullableDirectives_DifferentDirectives(string firstDirective, string secondDirective)
+        {
+            var tree1 = SyntaxFactory.ParseSyntaxTree($@"
+{firstDirective}
+class C
+{{
+}}");
+            var tree2 = SyntaxFactory.ParseSyntaxTree($@"
+{secondDirective}
+class C
+{{
+}}");
+
+            VerifyNotEquivalent(tree1, tree2, topLevel: true);
+            VerifyNotEquivalent(tree1, tree2, topLevel: false);
+            VerifyEquivalent(tree1.GetRoot(), tree2.GetRoot(), ignoreChildNode: k => k == SyntaxKind.NullableDirectiveTrivia);
+        }
+
+        [Theory, WorkItem(38694, "https://github.com/dotnet/roslyn/issues/38694")]
+        [InlineData("#nullable enable", "#nullable disable")]
+        [InlineData("#nullable enable", "#nullable restore")]
+        [InlineData("#nullable disable", "#nullable restore")]
+        [InlineData("", "#nullable disable")]
+        [InlineData("", "#nullable enable")]
+        [InlineData("", "#nullable restore")]
```

Comment:
can we have hte reverse as well?  i.e. the directive there before, but not afterwards?